packages/ember-data/lib/system/store/serializer-response.js

```diff
@@ -16,7 +80,8 @@ const get = Ember.get;
 */
 export function normalizeResponseHelper(serializer, store, modelClass, payload, id, requestType) {
   let normalizedResponse = serializer.normalizeResponse(store, modelClass, payload, id, requestType);
-
+  let validationErrors = validateDocumentStructure(normalizedResponse);
```

Comment:
this call shoul happen in the assert block, or it wont get stripped for prod builds